### PR TITLE
Fix detached

### PR DESCRIPTION
**Pull request by jollyromp**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -58,7 +58,7 @@ function loadConfig () {
 
 function getEnvIdFromBranch () {
   try {
-    let branch = sh.exec('git status', { silent: true }).stdout
+    var branch = sh.exec('git status', { silent: true }).stdout
```
Comment (andyday) [inline index.js]: this should be let not var

File: index.js [hunk 2]
```diff
@@ -71,10 +71,15 @@ function getEnvIdFromBranch () {
...
-    return _.trimEnd(_.truncate(branch, {
+    branch = _.trimEnd(_.truncate(branch, {
       length: 13,
       omission: ''
     }), '-')
+
+    var hash = sh.exec('git rev-parse HEAD').stdout
+    hash = hash.substring(0, 5)
+
+    return (branch.includes(' ')) ? hash : branch
```
Comment (andyday) [inline index.js]: if the branch includes spaces, I would prefer to replace them with dashes than to use the hash.  the hash is not user friendly

**Comment by oifland** [comment]
Andy, we're trying to catch the 'dettached head' scenario. Maybe we should
explicitly call that out?

**Comment by andyday** [comment]
@oifland are you doing this for it to work on gitlab ci?  If so you can pass in a --env=$CI_COMMIT_REF_NAME ??  Using the hash is problematic because it would mean a new environment for each commit...

**Comment by jollyromp** [comment]
Actually, wouldn't it be better if we used "git name-rev HEAD --name-only" instead of trying to parse git status?

**Comment by oifland** [comment]
Yes, we could do that instead. Right now we get weird errors when there's a
space in the name, so maybe it makes sense to make it clear and just refer
to the env as "detatched-head" and emit a warning to indicate your tip?

**Comment by andyday** [comment]
@jollyromp yes -- great callout.  `git name-rev HEAD --name-only` is much better than the current method, please make this fix.

@oifland as for spaces in names, we should either replace spaces with "" or with "-"

**Comment by jollyromp** [comment]
I added the git name stuff

**Comment by jollyromp** [comment]
also branches aren't allowed to have any spaces

File: index.js [hunk 3]
```diff
@@ -58,23 +58,14 @@ function loadConfig () {
...
-      branch = branch.replace(new RegExp(_.trim(config.branchRegex)), '$1')
-    }
-
-    return _.trimEnd(_.truncate(branch, {
+    branch = _.trimEnd(_.truncate(branch, {
       length: 13,
       omission: ''
     }), '-')
+
+    return branch
```
Comment (andyday) [inline index.js]: no need to have the separate variable.  you can return at line 63 directly

**Comment by andyday** [comment]
👍 LGTM!  One minor comment...

**Comment by andyday** [comment]
published as 0.10.0

**Comment by jollyromp** [comment]
How does this get uploaded to npm?

**Comment by andyday** [comment]
manually using npm publish.  I use a tool called np to make it a bit easier.

**Pull request merged**